feat: Account overview component

ibrkhalil opened the pull request:
fixes #16726 

### Summary
Figma on top

<img src="https://github.com/status-im/status-mobile/assets/33176106/3fa9044a-4ecf-488a-a578-789021d64a01"/> 

status: ready

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | ff37c90b | [#2](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-16742/2/) | 2023-07-21 10:27:08 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-16742/2/consoleText) |

J-Son89 commented on src/quo2/components/wallet/account_overview/component_spec.cljs:
```diff
@@ -0,0 +1,36 @@
+  (h/test "renders correct account value"
+    (h/render [account-overview/account-overview
+               {:current-value "20.0$"}])
+    (-> (js/expect (h/get-by-text "20.0$"))
```
@ulisesmac added a nice helper, h/is-truthy. There are some examples you can follow to use it :)

ibrkhalil replied:
Done

J-Son89 commented on src/quo2/components/wallet/account_overview/view.cljs:
```diff
@@ -0,0 +1,69 @@
+    (when show-details?
+      [icons/icon :i/reveal
+       {:container-style style/reveal-icon
+        :color           (colors/theme-colors colors/neutral-80-opa-40 colors/white-opa-40)}])]
```
if you're using `color/theme-colors` you need `theme` as the 3rd param. Eventually we will remove the internal checks in that function to the global ratom for theme.

so then you'll also need to wrap this component in the `with-theme` hoc

ibrkhalil replied:
Done

J-Son89 commented on src/quo2/components/wallet/account_overview/style.cljs:
```diff
@@ -0,0 +1,56 @@
+   :margin-top  2})
+
+(def current-value
+  {:color         (colors/theme-colors colors/neutral-100 colors/white)
```
is this not just the default values for text anyway?

ibrkhalil replied:
Cleaned

J-Son89 commented on src/quo2/components/wallet/account_overview/style.cljs:
```diff
@@ -0,0 +1,56 @@
+
+(def account-name
+  {:margin-bottom 3
+   :color         (colors/theme-colors colors/neutral-100 colors/white)})
```
same here? I think these are the default colors 🤔

ibrkhalil replied:
Cleaned

J-Son89 commented on src/quo2/components/wallet/account_overview/view.cljs:
```diff
@@ -0,0 +1,69 @@
+           percentage-change currency-change up?]}]
+  [rn/view
+   {:style style/account-overview-wrapper}
+   [rn/view style/row
```
can you instead do `[rn/view {:style style/row} ...` ?

ibrkhalil replied:
Done

J-Son89 commented on src/quo2/core.cljs:
```diff
@@ -257,3 +258,6 @@
 (def url-preview-list quo2.components.links.url-preview-list.view/view)
 (def link-preview quo2.components.links.link-preview.view/view)
+
+;;;; WALLET
```
nice!

ibrkhalil replied:
🙌

J-Son89 commented on src/quo2/components/wallet/account_overview/view.cljs:
```diff
@@ -0,0 +1,69 @@
+            [quo2.components.wallet.account-overview.style :as style]))
+
+(defn account-overview
+  [{:keys [account-name current-value bottom-time-text bottom-time-to-text show-details?
```
Hey @ibrkhalil, we want to move away from creating our own api names for components - it makes it very difficult to find the correct configuration when using the component on a given page and then also makes it arduous to verify the component is using the correct configuration. 
the api should be mostly reflecting this:
<img width="329" alt="Screenshot 2023-07-21 at 13 41 57" src="https://github.com/status-im/status-mobile/assets/22799766/0ddd6b2e-2382-45bc-919e-5d537a3150f5">

J-Son89 replied:
something like this will be easier to find imo:
```
"
 account :watched-address/ :default
 time-frame nil/:selected/:1-week/1-month/3-months/1-year/:all-time/:custom
 metrics :positive/:negative
 "

(defn account-overview
  [{:keys [account-name loading? account metrics time-frame time-frame-start-text time-frame-to-text
           current-value
           percentage-change currency-change]}]
  (let [bottom-time-text (cond
                           (or (= time-frame :selected) (= time-frame :custom)) time-frame-start-text
                           (= time-frame :1-week) "1 week"
                           (= time-frame :1-month) "1 month"
                           (= time-frame :3-months) "3 months"
                           (= time-frame :1-year) "1 year"
                           (= time-frame :all-time) "All time"
                           :else nil)
        bottom-time-to-text (when (= time-frame :custom) time-frame-to-text)]
  [rn/view
   {:style style/account-overview-wrapper}
   [rn/view style/row
    [quo2/text
     {:weight :medium
      :size   :paragraph-1
      :style  style/account-name}
     account-name]
    (when (= account :watched-address)
      [icons/icon :i/reveal
       {:container-style style/reveal-icon
        :color           (colors/theme-colors colors/neutral-80-opa-40 colors/white-opa-40)}])]
   [quo2/text
    {:weight :semi-bold
     :size   :heading-1
     :style  style/current-value}
    current-value]
   [rn/view
    {:style style/row-centered}
    [:<>
     [quo2/text
      {:weight :medium
       :size   :paragraph-2
       :style  (style/bottom-time-text bottom-time-to-text)}
      (if (seq bottom-time-text)
        bottom-time-text
        (i18n/label :t/all-time))]
     (when (seq bottom-time-to-text)
       [:<>
        [icons/icon :i/positive-right
         {:size            16
          :container-style style/right-arrow}]
        [quo2/text
         {:weight :medium
          :size   :paragraph-2
          :style  style/bottom-time-to-text}
         bottom-time-to-text]])]
    (when (and (seq percentage-change)
               (seq currency-change))
      [:<>
       [quo2/text
        {:weight :medium
         :size   :paragraph-2
         :style  style/percentage-change}
        percentage-change]
       [rn/view
        {:style style/dot-separator}]
       [quo2/text
        {:weight :medium
         :size   :paragraph-2
         :style  style/currency-change}
        currency-change]
       [icons/icon
        (if (= metrics :positive)
          :i/positive
          :i/negative)
        {:color (colors/custom-color-by-theme :blue 50 60)
         :size  16}]])]]))
```
cc @mmilad75, @erikseppanen as it was something recently we have been discussing 👍

J-Son89 replied:
also the loading variant should be added - is that in here too? 🤔

ibrkhalil replied:
By loading state you mean the skeleton one?

J-Son89 replied:
yep, see here:
<img width="344" alt="Screenshot 2023-07-21 at 14 34 09" src="https://github.com/status-im/status-mobile/assets/22799766/3e9471e5-96c5-4c96-b3ae-081c54730d81">

ibrkhalil replied:
Got it, I missed it.
I'll ping you when I need another look
Thanks Jamie

ibrkhalil replied:
Please take another look when you have time Jamie

J-Son89 commented:
very much a nit, but some devs have asked that we use branches with some description attached, e.g `feat/add-wallet-account-overview`

J-Son89 commented on src/quo2/components/wallet/account_overview/style.cljs:
```diff
@@ -0,0 +1,56 @@
+
+(defn bottom-time-text
+  [bottom-time-to-text]
+  {:color        (colors/theme-colors colors/neutral-80-opa-40 colors/white-opa-40)
```
same here for use of this function they need theme as a param now

ibrkhalil replied:
Cleaned

erikseppanen commented on src/quo2/components/wallet/account_overview/view.cljs:
```diff
@@ -0,0 +1,128 @@
+                                  :one-year     (i18n/label :t/one-year)
+                                  :all-time     (i18n/label :t/all-time)
+                                  :custom       time-frame-string)
+        up?                     (if (= metrics :positive)
```
`if ... true false` could be removed here, leaving just `(= metrics :positive)` which returns true/false anyway

J-Son89 commented on src/quo2/components/wallet/account_overview/view.cljs:
```diff
@@ -0,0 +1,128 @@
+  [{:keys [account time-frame time-frame-string time-frame-to-string account-name current-value
+           percentage-change currency-change theme metrics loading?]}]
+
+  (let [time-frame-string       (case time-frame
```
It needs a default case, right @ulisesmac ? 🤔

J-Son89 replied:
> It needs a default case, right @ulisesmac ? 🤔

Also what about :selected timeframe?

ulisesmac replied:
@J-Son89 @ibrkhalil 
Yes, please, we want to always define the default case for `case`, since it will throw an exception if a value is not considered :disappointed:

J-Son89 commented on src/quo2/components/wallet/account_overview/view.cljs:
```diff
@@ -0,0 +1,128 @@
+        up?                     (if (= metrics :positive)
+                                  true
+                                  false)
+        dark?                   (= :dark theme)
```
You can use theme-color function and the 3rd parameter is theme. 
e.g '(colors/theme-colors light-variant dark-variant theme)'

I would encourage to do that and remove all these variables for color names 👍

J-Son89 commented on src/quo2/components/wallet/account_overview/style.cljs:
```diff
@@ -0,0 +1,86 @@
+                   8)})
+
+(def right-arrow
+  {:margin-top   1.25
```
Why 1.25? I thought we always use integers?

ibrkhalil replied:
To make match design as closest as I could.

flexsurfer replied:
yeah it must be integer

J-Son89 commented on src/quo2/components/wallet/account_overview/view.cljs:
```diff
@@ -0,0 +1,128 @@
+            [react-native.core :as rn]
+            [utils.i18n :as i18n]
+            [quo2.components.wallet.account-overview.style :as style]
+            [quo2.theme :as theme]))
```
theme will clash with the 'theme' prop. Safer to use ':as quo.theme'

J-Son89 commented on src/quo2/components/wallet/account_overview/view.cljs:
```diff
@@ -0,0 +1,128 @@
+              :size  16}]])]])]))
+
+(def account-overview
+  "Create a account-overview UI component.
```
Better to include on the internal where the api is

ibrkhalil replied:
But we want it to be available directly at the consumer's side.

It'll show on the IDE when hovering over it.

J-Son89 replied:
Ah okay, sounds good 👌

ilmotta replied:
> But we want it to be available directly at the consumer's side.
>
> It'll show on the IDE when hovering over it.

This is not the case unfortunately @J-Son89 @ibrkhalil.

The vast majority of components are used outside `src/quo2` (all status-im2 namespaces), not from inside. We re-export vars in `quo2.core`, but this means the docstrings are NOT copied from the original vars. This is a known drawback with an easy solution in Clojure JVM, but a convoluted solution in ClojureScript (see below).

Therefore, currently, to get the docstring of a component, the dev needs to jump to `quo2.core` and from there look up the actual var with the docstring, or open the component file directly to read the docstring. Maybe some REPL ninjas do differently, but that's still not ideal.

Example output from CIDER showing public vars for consumers of the quo lib don't have metadata:

```
quo2.core
  Show: All
  Hide: Private Test Macro Function Var (0 items filtered)
  Group-by: Type Visibility 

  account (-) ·························· Not documented.
  account-avatar (-) ··················· Not documented.
  account-selector (-) ················· Not documented.
  action-drawer (-) ···················· Not documented.
  activity-log (-) ····················· Not documented.
  activity-logs-photos (-) ············· Not documented.
  add-reaction (-) ····················· Not documented.
  audio-tag (-) ························ Not documented.
  author (-) ··························· Not documented.
  banner (-) ··························· Not documented.
  browser-input (-) ···················· Not documented.
  button (-) ··························· Not documented.
  category (-) ························· Not documented.
  channel-actions (-) ·················· Not documented.
  etc
  etc
```

The library https://github.com/JeremS/dolly is promising and offers a `def-clone` macro that copies the doc metadata from one var to another (in another ns). This would solve all our problems, since both namespaces inside & outside `src/quo2` would have access to doc metadata. It's worth exploring, might be a quick win.

Otherwise, for the moment @ibrkhalil's suggestion is the best (although it solves only ~20% of the problem).

I'm cc'ing @ulisesmac because I know he's one of the devs that appreciate docstrings in quo components.

J-Son89 commented on src/quo2/components/wallet/account_overview/component_spec.cljs:
```diff
@@ -0,0 +1,36 @@
+    (-> (h/get-by-text "20.0$")
+        h/is-truthy))
+
+  (h/test "renders correct account value from date to date"
```
It would be nice to have each variant of time-frame etc covered with tests

ibrkhalil replied:
Will do

flexsurfer commented on src/quo2/components/wallet/account_overview/style.cljs:
```diff
@@ -0,0 +1,86 @@
+(ns quo2.components.wallet.account-overview.style)
+
+(def account-overview-wrapper
+  {:width           335
```
:width never should be used, paddings instead

J-Son89 replied:
But also let's not use margin on the outer container of a component. Better to just pass a prop 'container-style' so we can set it on the page where its being used 👍

ibrkhalil replied:
Can you please take another look @J-Son89

ulisesmac commented on src/status_im2/contexts/quo_preview/wallet/account_overview.cljs:
```diff
@@ -0,0 +1,92 @@
+                             :account              :default})]
+    (fn []
+      [rn/touchable-without-feedback {:on-press rn/dismiss-keyboard!}
+       [rn/view {:padding-bottom 150}
+        [rn/view {:flex 1}
+         [preview/customizer state descriptor]]
+        [rn/view
+         {:padding-vertical 60
+          :flex-direction   :row
+          :justify-content  :center}
+         [quo2/account-overview @state]]]])))
+
+(defn preview-account-overview
+  []
+  [rn/view
+   {:background-color (colors/theme-colors colors/white
+                                           colors/neutral-95)
+    :flex             1}
+   [rn/flat-list
+    {:flex                         1
+     :keyboard-should-persist-taps :always
+     :header                       [cool-preview]
+     :key-fn                       str}]])
```
@ibrkhalil  Please add the style properties inside the `:style` key for this namespace too

ibrkhalil replied:
That's the preview namespace, And not the component
I don't think we follow that convention here

ulisesmac replied:
@ibrkhalil The convention is used in all namespaces, it is not related to production code, it's related to code readability

J-Son89 replied:
I think it's the same confusion I had before. To be clear, can you put it in a ':style' key and not in a style.cljs file :)

ibrkhalil replied:
Sorry I got confused and thought something else, I totally agree
Doing it now!

flexsurfer commented on src/quo2/components/wallet/account_overview/view.cljs:
```diff
@@ -0,0 +1,124 @@
+(defn- account-overview-internal
+  [{:keys [account time-frame time-frame-string time-frame-to-string account-name current-value
+           percentage-change currency-change theme metrics loading?]}]
+
```
extra line

J-Son89 commented on src/quo2/components/wallet/account_overview/view.cljs:
```diff
@@ -0,0 +1,123 @@
+             {:color positive-icon-color
+              :size  16}]])]])]))
+
+(def account-overview
```
btw aligning with the best practices doc we should export this as `view`. So we can use `view` and `view-internal` here 👍

ibrkhalil replied:
In progress

ibrkhalil replied:
Done

J-Son89 commented on src/quo2/components/wallet/account_overview/view.cljs:
```diff
@@ -0,0 +1,123 @@
+        positive-icon-color     (colors/theme-colors (colors/custom-color :blue 50)
+                                                     (colors/custom-color :blue 60)
+                                                     theme)
+        account-name-color      (colors/theme-colors colors/neutral-100 colors/white theme)
```
these variable colors make this file large and hard to read - why not just set these colors where they are used?

ibrkhalil replied:
Refactored

J-Son89 commented on src/status_im2/contexts/quo_preview/wallet/account_overview.cljs:
```diff
@@ -0,0 +1,92 @@
+(ns status-im2.contexts.quo-preview.wallet.account-overview
+  (:require [quo2.components.wallet.account-overview.view :as quo2]
```
import from `quo2.core` we want to treat `quo2` components as it's own module. i,e any uses of it outside of that scope should come from `quo2.core` 👍

J-Son89 commented on src/status_im2/contexts/quo_preview/main.cljs:
```diff
@@ -101,6 +101,7 @@
     [status-im2.contexts.quo-preview.wallet.network-amount :as network-amount]
     [status-im2.contexts.quo-preview.wallet.network-breakdown :as network-breakdown]
     [status-im2.contexts.quo-preview.wallet.token-overview :as token-overview]
+    [status-im2.contexts.quo-preview.wallet.account-overview :as account-overview]
```
this file is large, generally these imports are alpha sorted to help make it easier to use, let's keep it that way :)

J-Son89 commented on src/quo2/components/wallet/account_overview/component_spec.cljs:
```diff
@@ -0,0 +1,77 @@
+               {:currency-change   "€0.00"
+                :percentage-change "0.00%"}])
+    (and
+     (-> (h/get-by-text "0.00%")
```
do we need this `->` here?
won't `(h/is-truthy (h/get-by-text "0.00%"))` be sufficient here? same for most of these tests? 🤔

J-Son89 commented on src/quo2/components/wallet/account_overview/component_spec.cljs:
```diff
@@ -0,0 +1,67 @@
+                            {:time-frame        :one-week
+                             :currency-change   "€0.00"
+                             :percentage-change "0.00%"}])
+                 (h/is-truthy (h/get-by-text (i18n/label :t/one-week-int))))
```
oh I believe we don't want to use `i18n` directly in tests - 🤔 instead we just use the key for the translation string - please see:
https://github.com/status-im/status-mobile/blob/develop/src/quo2/components/dividers/strength_divider/component_spec.cljs#L12

J-Son89 commented on src/quo2/components/wallet/account_overview/view.cljs:
```diff
@@ -0,0 +1,116 @@
+                              time-frame-string)
+        up?                 (= metrics :positive)
+        reveal-icon-color   (colors/theme-colors colors/neutral-80-opa-40 colors/white-opa-40 theme)
+        positive-icon-color (colors/theme-colors (colors/custom-color customization-color 50)
```
this icon color is used for both `positive` and `negative` no? should just be `icon-color`? also you are using it for text colors too? it's a bit confusing/misleading

ibrkhalil replied:
It used the same value, So I though I'd reuse it

ibrkhalil replied:
It used the same value, So I thought I'd reuse it.
But you're right, Fixing

J-Son89 replied:
Oh yeah that's fine, it's just the name should be appropriate for the use case imo - even just custom-color or something is good 👌

OmarBasem commented on src/quo2/components/wallet/account_overview/view.cljs:
```diff
@@ -0,0 +1,122 @@
+  [{:keys [account time-frame time-frame-string time-frame-to-string account-name current-value
+           percentage-change currency-change theme metrics loading? customization-color]
+    :or   {customization-color :blue}}]
+  (let [time-frame-string (case time-frame
+                            :one-week     (i18n/label :t/one-week-int)
+                            :one-month    (i18n/label :t/one-month-int)
+                            :three-months (i18n/label :t/three-months-int)
+                            :one-year     (i18n/label :t/one-year)
+                            :all-time     (i18n/label :t/all-time)
+                            time-frame-string)
```
`time-frame-string` can be extracted outside of component

ibrkhalil replied:
Done

OmarBasem commented on src/quo2/components/wallet/account_overview/view.cljs:
```diff
@@ -0,0 +1,122 @@
+             time-frame-string)]
+          (when (and (= :custom time-frame)
+                     (seq time-frame-to-string))
+            [:<>
+             [icons/icon :i/positive-right
+              {:size            16
+               :container-style style/right-arrow}]
+             [quo2/text
+              {:weight :medium
+               :size   :paragraph-2
+               :style  style/bottom-time-to-text}
+              time-frame-to-string]])]
+         (when (and (seq percentage-change)
+                    (seq currency-change))
+           [:<>
+            [quo2/text
+             {:weight :medium
+              :size   :paragraph-2
+              :style  (style/percentage-change (colors/theme-colors
+                                                (colors/custom-color customization-color 50)
+                                                (colors/custom-color customization-color 60)
+                                                theme))}
+             percentage-change]
+            [rn/view
+             {:style (style/dot-separator (colors/theme-colors
+                                           (colors/custom-color customization-color 60 40)
+                                           (colors/custom-color customization-color 50 40)
+                                           theme))}]
+            [quo2/text
+             {:weight :medium
+              :size   :paragraph-2
+              :style  (style/currency-change (colors/theme-colors
+                                              (colors/custom-color customization-color 50)
+                                              (colors/custom-color customization-color 60)
+                                              theme))}
+             currency-change]
+            [icons/icon
+             (if up?
+               :i/positive
+               :i/negative)
```
@ibrkhalil The component tree is too big. Better break it up into smaller components.

@ilmotta @J-Son89 What do you guys think of adding a tree `max-lines-length` to our guidelines? (or maybe is it possible to be added to linter?)

ibrkhalil replied:
I think we can have a `clj-kondo` custom hook that detects it and shows a linting error.

J-Son89 replied:
I agree that probably it's nice to break this into smaller pieces but so sure we need a fixed rule for this. 
No strong opinions but I think we all have a good enough sense of when something can be broken down smaller.

ibrkhalil replied:
There's a lot of fun stuff [here](https://github.com/clj-kondo/clj-kondo/blob/master/doc/linters.md)

ilmotta replied:
There are reasons this kind of rule is usually not implemented/used in React projects. There's a spectrum of tree sizes that are acceptable and that eventually become too big. I agree with @J-Son89, some things are better if we create awareness in the team, since there's no definite tree size rule, it's more about React best practices.

ibrkhalil replied:
Refactored in 7491581fbcc9b63bf76ea50b5e550b8a88a33c01

ilmotta commented on src/quo2/components/wallet/account_overview/view.cljs:
```diff
@@ -0,0 +1,122 @@
+(defn- loading-state
+  [color]
+  [:<>
+   [rn/view (style/loading-bar-margin-bottom color 104 14)]
```
I find hard to read code like this, where "random" numbers are written. Better would be to convert the args to a map with `:color`, `:width` and `:height` keys.

I've seen some discussions where the argument is that "up to 3-4 args is fine to use positional args". Nothing wrong with this recommendation, which I've seen I think in a very old Java book from the 90s (extract 4 or more arguments into a separate class and blablabla). I think it's not that clear cut, readability also counts.

ilmotta commented on src/status_im2/contexts/quo_preview/wallet/account_overview.cljs:
```diff
@@ -0,0 +1,100 @@
+   {:label "Account name"
+    :key   :account-name
+    :type  :text}
+   {:label   "Customization color:"
```
It would be nice if we stopped copy & pasting this customization color field. It doesn't need to be the perfect code, just something to keep the preview code more DRY. For instance, extract the entire map generation to `status-im2.contexts.quo-preview.preview`.

ibrkhalil replied:
Happy to do it in a followup! :)

ilmotta commented on src/status_im2/contexts/quo_preview/wallet/account_overview.cljs:
```diff
@@ -0,0 +1,100 @@
+   {:label "Current value"
+    :key   :current-value
+    :type  :text}
+   {:label "Loading"
```
"Loading" -> "Loading?"

ilmotta commented on src/quo2/components/wallet/account_overview/view.cljs:
```diff
@@ -0,0 +1,134 @@
+                                             colors/white-opa-40
+                                             theme)}])])
+
+(defn- account-overview-internal
```
Can be just `view-internal`.

ilmotta commented on src/quo2/components/wallet/account_overview/component_spec.cljs:
```diff
@@ -0,0 +1,66 @@
+      (h/render [account-overview/view
+                 {:time-frame-string    "15 May"
+                  :time-frame-to-string "19 May"}])
+      (and
```
`and` shouldn't be used because the assertion can either fail or pass. `h/is-truthy` is an assertion. If they fail they short-circuit the test.

ibrkhalil replied:
Where I used `and` in the test, _always_ showed together in their UI.

ilmotta replied:
I didn't quite understand your last message @ibrkhalil.

What I mean is that `h/is-truthy` is a final assertion, there's no meaningful return value for component specs, so `and` is doing nothing here. The function is named `is-truthy` to resemble the `cljs.test/is` macro`, which fails the test on falsey values.

ibrkhalil replied:
Oh, Apologies
It'll always short-circuit if a value is falsey.
Forgot that :)

ibrkhalil replied:
Oh, Apologies
It'll always short-circuit if a value is falsey.
Forgot that :)

ilmotta commented on src/quo2/components/wallet/account_overview/view.cljs:
```diff
@@ -0,0 +1,134 @@
+           [numeric-changes percentage-change currency-change customization-color theme up?])]])]))
+
+(def view
+  "Create a account-overview UI component.
```
Shouldn't the component be receiving a `state` parameter to match Figma? Why the exception for this Figma property if we're mirroring other properties?

I don't see a practical advantage to flatten the `state` property into boolean flags. The code is still simple enough if it uses the same names as Figma. It also helps developers create a preview screen that mirrors exactly what's in Figma (useful for designers too).

ibrkhalil replied:
Fixing.

ilmotta commented on src/quo2/components/wallet/account_overview/view.cljs:
```diff
@@ -0,0 +1,134 @@
+  | `:time-frame`        | A Keyword that represents the type of the timeframe, Can be from a preset of time frames. `[:1-week :1-month :3-months :1-year :custom]` (default `nil`) if custom is set, We expect a start time and if there's a space between two times that are in `time-frame-string` we'll split them with an arrow to the right icon
+  | `:loading?`          | A Boolean that indicates that component is loading data. (default `nil`)
+  | `:metrics`           | A Keyword that indicates if the account value have increased can be `:positive` or `:negative` (default `nil`)
+  | `:time-frame-string` | A String representing time frame string. (default `nil`)
```
It's not idiomatic in Clojure to name things with their type in the name.

ibrkhalil replied:
Fixed

ilmotta commented on src/quo2/components/wallet/account_overview/view.cljs:
```diff
@@ -0,0 +1,148 @@
+(ns quo2.components.wallet.account-overview.view
+  (:require [quo2.components.icon :as icons]
+            [quo2.components.markdown.text :as quo2]
```
This ns alias doesn't seem right, shouldn't it be just `text`?

ibrkhalil replied:
Done

ilmotta commented on src/status_im2/contexts/quo_preview/wallet/account_overview.cljs:
```diff
@@ -0,0 +1,104 @@
+(ns status-im2.contexts.quo-preview.wallet.account-overview
+  (:require [quo2.core :as quo2]
```
Avoid aliasing to `quo2` unless strictly necessary. This will save us the trouble of changing lots of references from `quo2` to `quo` in the future (once we delete the old lib).

erikseppanen commented on src/quo2/components/wallet/account_overview/style.cljs:
```diff
@@ -0,0 +1,88 @@
+   :margin-top       8
+   :width            2
+   :height           2
+   :border-radius    200})
```
`:border-radius` not needed here, I think

ibrkhalil replied:
Reverted

OmarBasem commented on src/quo2/components/wallet/account_overview/view.cljs:
```diff
@@ -0,0 +1,148 @@
+  [:<>
+   [quo2/text
+    {:weight :medium
+     :size   :paragraph-2
+     :style  (style/percentage-change (colors/theme-colors
+                                       (colors/custom-color customization-color 50)
+                                       (colors/custom-color customization-color 60)
+                                       theme))}
+    percentage-change]
+   [rn/view
+    {:style (style/dot-separator (colors/theme-colors
+                                  (colors/custom-color customization-color 60 40)
+                                  (colors/custom-color customization-color 50 40)
+                                  theme))}]
+   [quo2/text
+    {:weight :medium
+     :size   :paragraph-2
+     :style  (style/currency-change (colors/theme-colors
+                                     (colors/custom-color customization-color 50)
+                                     (colors/custom-color customization-color 60)
+                                     theme))}
+    currency-change]
+   [icons/icon
+    (if up?
+      :i/positive
+      :i/negative)
+    {:color (colors/theme-colors (colors/custom-color customization-color 50)
+                                 (colors/custom-color customization-color 60)
+                                 theme)
+     :size  16}]])
```
I think these `colors/custom-color` should be called from the style file. Also they are repeated 4 times -- they can be extracted into one definition

ibrkhalil replied:
I did, but @J-Son89 thought using the same variable name for two style props was misleading so I reverted it.

ibrkhalil replied:
Done

ibrkhalil committed:
Quo2: Account overview component

ibrkhalil committed:
Fix tests

ibrkhalil committed:
Merge branch 'develop' into 16726

ibrkhalil commented:
Can I get a final look here?
@J-Son89 @OmarBasem

ibrkhalil committed:
Merge branch 'develop' into 16726

J-Son89 commented on src/quo2/components/wallet/account_overview/component_spec.cljs:
```diff
@@ -0,0 +1,65 @@
+
+  (h/test "renders correct account timeframe"
+    (h/render [account-overview/view
+               {:time-frame        :all-time
```
nit: maybe we should also add a test for custom time frame? 🤔

ibrkhalil replied:
It's on line 24

J-Son89 commented on src/status_im2/contexts/quo_preview/main.cljs:
```diff
@@ -424,6 +425,9 @@
                        {:name      :network-amount
                         :options   {:topBar {:visible true}}
                         :component network-amount/preview}
+                       {:name      :account-overview
```
we should keep files like these alpha sorted to keep it organised. They're naturally quite large and it's easier to maintain this way 👍

ibrkhalil replied:
Done

J-Son89 reviewed (approved):
nice work @ibrkhalil, thanks for putting in the extra work to make the adjustments! 💪

Francesca-G reviewed (approved):
Here's the [Figma frame](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=3446%3A227419&mode=design&t=cN7t8rDQdERrKPEM-1) with the design review, approving with minor fixes in spacing and colors required in follow up :)

ibrkhalil committed:
Merge branch 'develop' into 16726

ibrkhalil committed:
Sort file

ibrkhalil committed:
Address feedback

ibrkhalil commented:
> Here's the [Figma frame](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=3446%3A227419&mode=design&t=cN7t8rDQdERrKPEM-1) with the design review, approving with minor fixes in spacing and colors required in follow up :)

I addressed the feedback, Kindly take another look when you have time

ibrkhalil committed:
Merge branch 'develop' into 16726

status-im-auto commented:
#### :heavy_check_mark: [status-mobile/prs/android-e2e/PR-16742#35](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-16742/35/) :small_blue_diamond: ~6 min 55 sec :small_blue_diamond: 368c42ea :small_blue_diamond: [:package: android-e2e package](https://status-im-mobile-prs.ams3.digitaloceanspaces.com/StatusIm-Mobile-230809-115145-368c42-pr16742-x86.apk)

ibrkhalil committed:
Merge branch 'develop' into 16726

ibrkhalil committed:
Merge branch 'develop' into 16726
